File: PrimeiraAplicacao.Funcional/2_ConceitosFundamentais/PeriodoDeTempo.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _2_ConceitosFundamentais
{
    public class PeriodoDeTempo
    {
        public DateTime DataInicial { get; set; }
        public DateTime DataFinal { get; set; }

        public bool VerificarSeDataEstaEntreOPeriodo(DateTime data) =>
            data.CompareTo(DataInicial) >= 0
                && data.CompareTo(DataFinal) <= 0;
    }
}

File: PrimeiraAplicacao.Funcional/2_ConceitosFundamentais.Funcional/PeriodoDeTempo.fs
module PeriodoDeTempo
open System

type Periodo = { DataInicial: DateTime; DataFinal: DateTime }

let adicionarDias periodo dias = {
    DataInicial = periodo.DataInicial.AddDays dias
    DataFinal = periodo.DataFinal.AddDays dias
}

let verificarSeDataEstaEntreOPeriodo periodo (data: DateTime) =
    data.CompareTo periodo.DataInicial >= 0 &&
    data.CompareTo periodo.DataFinal <= 0

let exemploDeUsoExpressao numero = 
    let resultado =
        if numero % 2 = 0 then 2 else 0
    printf "%i" resultado
    